### PR TITLE
executor, util: fix wrong behavior of group_concat(distinct)

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve? 
fix issue #10098 

### What is changed and how it works?
Use EncodeBytes to encode string values in groupConcatDistinct.
Before this commit, string values are encoded with true values.
e.g. 
encode("1", "122") --> "1122"
encode("11", "22") --> "1222"
These are not comparable.

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported function/method change

Side effects

N/A

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: util/set/string_set.go [hunk 1]
```diff
@@ -31,3 +31,12 @@ func (s StringSet) Exist(val string) bool {
 func (s StringSet) Insert(val string) {
 	s[val] = struct{}{}
 }
+
+// InsertIfNotExist inserts `val` into `s` if it not exists in `s`.
+func (s StringSet) InsertIfNotExist(val string) bool {
```
Comment (zz-jason) [inline util/set/string_set.go]: is it faster than `Insert()`?

Comment (XuHuaiyu) [reply util/set/string_set.go]: I think not.
This function is just introduced for the cases that we need to check `Exists` first.

File: executor/aggfuncs/func_group_concat.go [hunk 2]
```diff
@@ -175,24 +179,23 @@ func (e *groupConcatDistinct) UpdatePartialResult(sctx sessionctx.Context, rowsI
...
 			}
+			p.encodeBytesBuffer = codec.EncodeBytes(p.encodeBytesBuffer, hack.Slice(v))
 			p.valsBuf.WriteString(v)
 		}
 		if isNull {
 			continue
 		}
-		joinedVals := p.valsBuf.String()
-		if p.valSet.Exist(joinedVals) {
+		if p.valSet.InsertIfNotExist(string(p.encodeBytesBuffer)) {
```
Comment (winoros) [inline executor/aggfuncs/func_group_concat.go]: Can there use `hack.String()`?

Comment (winoros) [inline executor/aggfuncs/func_group_concat.go]: can there use `hack.String()`?

File: util/set/string_set.go (hunk 1 above)
Comment (zz-jason) [reply util/set/string_set.go]: So I think we can replace this function with `Insert()`.

File: executor/aggfuncs/func_group_concat.go (hunk 2 above)
Comment (XuHuaiyu) [reply executor/aggfuncs/func_group_concat.go]: @winoros 
Noop.
p.encodeBytesBuffer will be re-used.

Comment (qw4990) [inline executor/aggfuncs/func_group_concat.go]: Why introduce this new function `InsertIfNotExist`?
How about just replace `joinedVals` with `string(encodeBytesBuffer)` when calling `Exist` and `Insert`.

**Pull request merged**